### PR TITLE
More api

### PR DESCRIPTION
**Pull request by pentacular**
This removes 'keep' and generalizes drop to use a selector -- e.g., keep('a') becomes drop(getNot('a')), while drop('a') becomes drop(get('a'))

deform is also simplified

The api nb style is changed a bit -- we'll need another pass through these at the end. :)

### CODE REVIEW

File: nb/api/in.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+### in()
+Produces the interior of an item.
```
Comment (BarbourSmith) [inline nb/api/in.md:2]: I think that people might read this as it gives the volume contained in the solid that is represented by the item. I know what you mean, and I'm not sure how else to say it, but I think first time users might not get it. Maybe "Produces the interior of an items tree"? Or something?

**Review by BarbourSmith — APPROVED** [review]
This is gorgeous. This is exactly what first time users need. I love it so much!!

File: api/shape/billOfMaterials.js [hunk 2]
```diff
@@ -1,8 +1,9 @@
 import Shape from './Shape.js';
+import note from './note.js';
 
 // Is this better than s.get('part:*').tags('part')?
 export const billOfMaterials = Shape.chainable(
-  (op = (list) => (shape) => shape.md(`Materials: ${list.join(', ')}`)) =>
+  (op = (list) => note(`Materials: ${list.join(', ')}`)) =>
```
[the remark this answers is not in the document]
Comment (pentacular) [reply api/shape/billOfMaterials.js:6]: Yeah, I eventually realized that md working at the top level like that meant that it couldn't integrate with Shape properly, so I put together an alternative that does.

File: api/shape/eachIn.js [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import Shape from './Shape.js';
 import { getLeafsIn } from '@jsxcad/geometry';
 
+// CHECK: Is this obsoleted by .in().each()?
```
[the remark this answers is not in the document]
Comment (pentacular) [reply api/shape/eachIn.js:4]: Yeah, I think I'll delete eachIn as I can't remember the last time I used it.

File: nb/api/Link.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+Constructs a polyline from the segments of shapes.
+
+The shapes are linked by straight segments.
+
+The polyline is not closed.
+
+See: [Loop](#https://raw.githubusercontent.com/jsxcad/JSxCAD/master/nb/api/Loop.nb)
+
+_Note: We should rethink Arc and Box being implicitly filled._
```
[the remark this answers is not in the document]
Comment (pentacular) [reply nb/api/Link.md:14]: I don't think so -- extrude can call fill optimistically.

File: nb/api/flat.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
+### flat()
+
+Puts the shape into its preferred orientation (which is not necessarily flat).
```
[the remark this answers is not in the document]
Comment (pentacular) [reply nb/api/flat.md:3]: If you can think of a better name. :)

File: nb/api/in.md (hunk 1 above)
Comment (pentacular) [reply nb/api/in.md:2]: Hmm, good point.

Comment (pentacular) [reply nb/api/in.md:2]: How about "Produces the shape structure inside the item." ?

File: nb/api/Link.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (pentacular) [reply nb/api/Link.md:14]: Ah, actually, I forgot about extruding segments -- extruding the outline of a box should get you four walls.

Yeah, you'd have to call fill, so Box(4).f().ez(1) or something.

Needs a bit more thought, I think.

File: api/shape/eachIn.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (BarbourSmith) [reply api/shape/eachIn.js]: I support that. I have used it, but I didn't know about .in().each() so I'll switch over

File: nb/api/Link.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (BarbourSmith) [reply nb/api/Link.md:14]: I do think Arc(10).ez(4) is really nice. Having to fill it seems like a cumbersome extra step maybe

File: nb/api/flat.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (BarbourSmith) [reply nb/api/flat.md:3]: 😬

File: nb/api/in.md (hunk 1 above)
Comment (BarbourSmith) [reply nb/api/in.md]: I like that. I think that the examples are really what makes it clear

File: nb/api/Link.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (pentacular) [reply nb/api/Link.md:14]: Yeah, it's also hard to see the surfaces. I guess it's about right at the moment.

**Pull request merged**